AutoscoperM/AutoscoperM.py

```diff
@@ -0,0 +1,689 @@
+import contextlib
+import os, time, subprocess
+import unittest
+import logging
+import vtk, qt, ctk, slicer
+from slicer.ScriptedLoadableModule import *
+from slicer.util import VTKObservationMixin
+
+slicer.util.pip_uninstall("PyAutoscoper") # need to uncomment this line if you want to upgrade the PyAutoscoper package
+try:  # from https://discourse.slicer.org/t/install-python-library-with-extension/10110/2
+    from PyAutoscoper.connect import AutoscoperConnection
+except:
+    slicer.util.pip_install("PyAutoscoper")
+    from PyAutoscoper.connect import AutoscoperConnection
```

Comment:
@jcfr is there a better way to ensure that the PyAutoscoper package stays up to date?

Reply:
After the relevant requirements are installed, the environment is expected to be stable and should not be updated each time the user restart the application.

This would for example prevent from using the software if not internet connection is available.

The time being I will remove the call to `slicer.util.pip_uninstall("PyAutoscoper")`

> package stays up to date?

Since we will ship autoscoper along side the application, specifying the version using the compatible release operator [^1] (e.g `slicer.util.pip_install("PyAutoscoper~=1.1.0")` is a reasonable tradeoff.


[^1]: https://peps.python.org/pep-0440/#compatible-release